src/drive.ts

```diff
@@ -1,7 +1,21 @@
 import { Contents, ServerConnection } from '@jupyterlab/services';
 
+import { PathExt } from '@jupyterlab/coreutils';
+
 import { ISignal, Signal } from '@lumino/signaling';
 
+async function toArray<T>(
```

Comment:
Wondering if the one from Lumino would work here?

```ts
import { toArray } from '@lumino/algorithm';
```

Reply:
If so we can then add `@lumino/algorithm` to the `dependencies` in `package.json`

Reply:
I didn't know this was a thing, thanks! :)

Reply:
Ah actually this does not work as the `toArray` I introduced takes and `AsyncIterableIterator` as input, while Lumino's takes an `ArrayLike` which is incompatible.

Reply:
Ah ok thanks for checking :+1: